packages/core/src/codewhisperer/commands/startSecurityScan.ts

```diff
@@ -250,7 +250,11 @@ export async function startSecurityScan(
                 CodeScansState.instance.setMonthlyQuotaExceeded()
             }
         }
-        codeScanTelemetryEntry.reason = (error as SecurityScanError).message
+        codeScanTelemetryEntry.reasonDesc =
+            (error as ToolkitError)?.code === 'ContentLengthError'
+                ? 'Payload size limit reached'
+                : getTelemetryReasonDesc(error)
+        codeScanTelemetryEntry.reason = (error as ToolkitError)?.code ?? 'DefaultError'
```

Comment:
you may want to use `getTelemetryReason`  https://github.com/aws/aws-toolkit-vscode/blob/6bc6cf13cddafe23d48e5340f16efc55b6e00f3a/packages/core/src/shared/errors.ts#L317